File: 0258-add-digits/0258-add-digits.c
int addDigits(int num) {
    if (num == 0)
        return 0;

    int sum = 0;

    while (num > 0) {
        sum += num % 10;
        num /= 10;
    }

    if (sum < 10)
        return sum;

    return addDigits(sum);
}

File: 0258-add-digits/0258-add-digits.py
class Solution:
    def addDigits(self, num: int) -> int:
        count = 0 
        for i in str(num):
            count += int(i)

        if len(str(count)) == 1:
            return int(count)
        else:
            return self.addDigits(count)      
            